src/themes/default/store/homepage.ts

```diff
@@ -1,6 +1,29 @@
+import {prepareQuery} from '@vue-storefront/core/modules/catalog/queries/common'
+
 export const homepageStore = {
   namespaced: true,
   state: {
     new_collection: []
+  },
+  actions: {
+    fetchNewCollection ({ commit, dispatch }) {
+      const newProductsQuery = prepareQuery({ queryConfig: 'newProducts' })
+
+      dispatch('product/list', {
```

Comment:
please just change this to async/await

Reply:
@patzick awaits are usually slower, that's why I avoid them. Dispatch should be returned though, I give you that.

Reply:
I also would like to see such studies since all I ever read was saying it's still a microtask under the hood and just syntax sugar for Promises ;)

It can be slower only if you don't know how it works (not concurrently) and expect multiple await to run simultaneously but it's exactly the same with Promises

Reply:
@patzick https://v8.dev/blog/fast-async it was this one